site/src/components/decorate-pattern.md

```diff
@@ -0,0 +1,229 @@
+
+<script type="text/javascript">
+(function() {
+  var f = createFixture('#bar-example', null, 200, 10, function() { return true; });
+  var container = f.container, data = f.data,
+    xScale = f.xScale, yScale = f.yScale;
+
+  {{{bar-example-code}}}
+
+  container.append('g')
+      .datum(data)
+      .call(bar);
+}());
+</script>
+
+Again, the enter selection is used and the required child element, in this case a `path` is selected. D3 selections allow you to specify values as functions of the bound data and index, here the index is used to cycle through a color scale.
+
+## Adding data labels
+
+Using the decorate pattern you can also add new elements to a component via its data join. The d3fc series are designed so that the containing `g` element is translated to the location of each datum, as a result any elements that you add need only have a relative transform applied.
+
+In the example below datapoint labels are added via decorate:
+
+```js
+{{{point-example-code}}}
+```
+
+<div id="label-example" class="chart" style="height: 200px"> </div>
+
+<script type="text/javascript">
+(function() {
+  var f = createFixture('#label-example', null, 200, 10, function() { return true; });
+  var container = f.container, data = f.data,
+    xScale = f.xScale, yScale = f.yScale;
+
+  {{{point-example-code}}}
+
+  container.append('g')
+      .datum(data)
+      .call(point);
+}());
+</script>
+
+
+## Enter and exit transitions
+
+The examples so far have all acted on the enter selection, which is the selection that is executed when new items are initially added to the bound data. Via data join it is also possible to add logic to the exit and update selections.
+
+In this (slightly contrived) example the mouse location is plotted via a point series. The exit and enter selections are used to add transitions to the circle radius:
+
+```js
+{{{enter-exit-example-code}}}
+```
+
+<div id="enter-exit-example" class="chart" style="height: 300px"> </div>
+
+<script type="text/javascript">
+(function() {
+
+  var width = 600, height = 300;
+  var container = d3.select('#enter-exit-example')
+        .insert('svg', 'div')
+        .attr('width', width)
+        .attr('height', height);
+  var series = container.append('g');
+
+  var xScale = d3.scale.linear()
+    .domain([0, 100])
+    .range([0, width]);
+
+  var yScale = d3.scale.linear()
+    .domain([0, 100])
+    .range([0, height]);
+
+  var data = [];
+
+  var mouseX = 50, mouseY = 50;
+
+  document.onmousemove = function(evt) {
+    mouseX = evt.clientX * 100 / $(window).width();
+    mouseY = evt.clientY * 100 / $(window).height();
+  }
+
+  function render() {
+
+    data.push({x: mouseX, y: mouseY});
+
+    if (data.length > 20) {
+      data = data.slice(1);
+    }
+
+    {{{enter-exit-example-code}}}
+
+      series.datum(data)
+        .transition()
+        .duration(500)
+        .call(point);
+  }
+
+  setInterval(render, 200);
```

Comment:
> requestAnimationFrame

I don't really want 60 circles added per second, were you thinking of `requestAnimationFrame` plus something like `if (tick%10 === 0)`?